openmls/src/group/mls_group/errors.rs

```diff
@@ -6,102 +6,97 @@
-            CryptoError(CryptoError) =
-                "See [`CryptoError`](openmls_traits::types::CryptoError) for details.",
-            WelcomeError(WelcomeError) = "See [`WelcomeError`] for details.",
-        }
-    }
+/// MlsGroup error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum MlsGroupError {
+    #[error(transparent)]
+    LibraryError(#[from] LibraryError),
+    #[error(
+        "Couldn't find a `CredentialBundle` in the `KeyStore` that matches the one in my leaf."
+    )]
+    NoMatchingCredentialBundle,
+    #[error("Couldn't find a `KeyPackageBundle` in the `KeyStore` that matches the given `KeyPackage` hash.")]
+    NoMatchingKeyPackageBundle,
+    #[error("Tried to access a poisoned `CredentialBundle`. See [`PoisonError`](`std::sync::PoisonError`) for details.")]
+    PoisonedCredentialBundle,
+    #[error("No signature key was available to verify the message signature.")]
+    NoSignatureKey,
+    #[error("Can't create a new commit while another commit is still pending. Please clear or merge the pending commit before creating a new one.")]
+    PendingCommitError,
+    #[error("There is no pending commit that can be merged.")]
+    NoPendingCommit,
+    #[error("Can't clear an external commit, as the group can't merge `Member` commits yet. If an external commit is rejected by the DS, a new external init must be performed. See the MLS spec for more information.")]
+    ExternalCommitError,
+    #[error("Error performing key store operation.")]
+    KeyStoreError,
+    #[error("The incoming message's wire format was not compatible with the wire format policy for incoming messages.")]
+    IncompatibleWireFormat,
+    #[error(transparent)]
+    Config(#[from] ConfigError),
+    #[error(transparent)]
+    Group(#[from] CoreGroupError),
+    #[error(transparent)]
+    CreateCommit(#[from] CreateCommitError),
+    #[error(transparent)]
+    GroupStateError(#[from] MlsGroupStateError),
+    #[error(transparent)]
+    Exporter(#[from] ExporterError),
+    #[error(transparent)]
+    EmptyInput(#[from] EmptyInputError),
+    #[error(transparent)]
+    CredentialError(#[from] CredentialError),
+    #[error(transparent)]
+    TreeSyncError(#[from] TreeSyncError),
+    #[error(transparent)]
+    ValidationError(#[from] ValidationError),
+    #[error(transparent)]
+    TlsCodecError(#[from] TlsCodecError),
+    #[error(transparent)]
+    CryptoError(#[from] CryptoError),
+    #[error(transparent)]
+    WelcomeError(#[from] WelcomeError),
 }
 
-implement_error! {
-    pub enum EmptyInputError {
-        AddMembers = "An empty list of KeyPackages was provided.",
-        RemoveMembers = "An empty list of indexes was provided.",
-    }
+/// EmptyInput error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum EmptyInputError {
+    #[error("An empty list of KeyPackages was provided.")]
+    AddMembers,
+    #[error("An empty list of indexes was provided.")]
```

Comment:
While you're at it ...
```suggestion
    #[error("An empty list of KeyPackage references was provided.")]
```